registry/auth.go

```diff
@@ -74,6 +74,13 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorizedV2() bool {
```

Comment:
I know this is nitpicking, but this code can be a bear to navigate. Perhaps, we call this `CanAuthorizeV2`?